web-app/src/app/screens/SignUp.tsx

```diff
@@ -105,7 +110,16 @@ export default function SignUp(): React.ReactElement {
     const provider = oathProviders[oauthProvider];
     signInWithPopup(auth, provider)
       .then((userCredential: UserCredential) => {
-        dispatch(loginWithProvider({ oauthProvider, userCredential }));
+        if (!userCredential.user.emailVerified) {
+          dispatch(verifyEmail());
+        }
+        if (userCredential.user.email == null) {
+          alert(
+            'No public email provided in Github account. Please use a different registration method.',
+          );
```

Comment:
[suggestion]: Use a Snackbar component with no _autoHideDuration_ set instead of JS alert.